File: src/main/webapp/src/app/replay/StepForwardCommand.ts
export class StepForwardCommand {

    type: string;
    runUntil: number;
    numSteps: number;

    constructor( type: string, runUntil: number, steps: number ) {
        this.type = type;
        this.runUntil = runUntil;
        this.numSteps = steps;
    }
}

File: src/main/java/com/zuehlke/fnf/utsukushii/replay/StepForwardCommand.java
package com.zuehlke.fnf.utsukushii.replay;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data @AllArgsConstructor
class StepForwardCommand {

    public enum Type {
        Steps,
        Until
    }

    Long runUntil;
    Integer numSteps;
    Type type;
}
